src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,111 @@
+        c.argument('conf', help='The configuration of spark batch job.')
+        c.argument('driver_memory', help='The memory of driver.')
+        c.argument('driver_cores', help='The number of cores in driver.')
+        c.argument('executor_memory', help='The memory of executor.')
+        c.argument('executor_cores', help='The number of cores in each executor.')
+        c.argument('num_executors', help='The number of executors.')
+
+    with self.argument_context('synapse spark batch') as c:
+        c.argument('batch_id', options_list=['--id'], arg_group='Spark Batch', help='The id of the spark batch job.')
+        c.argument('job_type', help='The spark batch job type: SCALA, PYTHON or DOTNET. The default value is SCALA.')
+
+    with self.argument_context('synapse spark session') as c:
+        c.argument('session_id', options_list=['--id'], arg_group='Spark Session',
+                   help='The id of the spark session job.')
+        c.argument('num_executors', help='The number of executors.')
+
+    with self.argument_context("synapse spark session-statement") as c:
+        c.argument('statement_id', options_list=['--id'], arg_group="Spark Session-statement",
+                   help='The id of the statement.')
+        c.argument('session_id', help='The id of spark session job.')
+        c.argument('code', help='The code of spark statement.')
+        c.argument('kind', help='The kind of spark statement.')
+
+    # synapse workspace
+    with self.argument_context('synapse workspace') as c:
+        c.argument('resource_group_name', help='The resource group name.')
+        c.argument('workspace_name', arg_type=name_type, help='The workspace name.')
+        c.argument("account_url", help='The data lake storage account url.')
+        c.argument('file_system', help='The file system of the data lake storage account.')
+        c.argument('sql_admin_login_user', help='The sql administrator login user name.')
+        c.argument('sql_admin_login_password', help='The sql administrator login password.')
+        c.argument('tags', arg_type=tags_type, help='The tag of workspace.')
+        c.argument('identity_type', help='The type of managed identity.')
+        c.argument('principal_id', help='The principal id of managed identity.')
+        c.argument('tenant_id', help='The tenant id of managed identity.')
+
+    # synapse spark pool
+    with self.argument_context('synapse spark pool') as c:
+        c.argument('spark_pool_name', arg_type=name_type, help='The spark pool name.')
+        c.argument('workspace_name', help='The workspace name.')
+
+        # Node
+        c.argument('node_count', arg_group='Node', help='The number of node.')
+        c.argument('node_size_family', arg_group='Node', help='The node size family.')
+        c.argument('node_size', arg_group='Node', help='The node size.')
+
+        # AutoScale
+        c.argument('auto_scale_enabled', arg_group='AutoScale', help='The flag of enabling auto scale.')

```

Comment:
recommend to use [get_three_state_flag](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-flags) for boolean properties.